src/sources/soundsourcem4a.cpp

```diff
@@ -631,7 +694,9 @@ ReadableSampleFrames SoundSourceM4A::readSampleFramesClamped(
                     m_inputBufferLength = inputBufferLength;
                     m_inputBufferOffset = 0;
                     // Skip ADTS header if we have a decoder specific ES config
-                    if (m_pMP4ESConfigBuffer &&
+                    AVCodecParameters* par = m_pAvStream->codecpar;
+                    DEBUG_ASSERT(par != nullptr);
+                    if (par->extradata && par->extradata_size > 0 &&
```

Comment:
Your usage of `!= nullptr` is inconsistent. Here you omit it whereas in other situations you add it.